### DISCUSSION TITLE
Creating mesh of dicretization sizes in the same file

### DISCUSSION THREAD

**Discussion by salaudeen-ya** [idaholab/moose#23914] [Q&A Meshing] [answered]
Hello,

I would like to have a mesh discretization like

![image](https://user-images.githubusercontent.com/38023107/228970482-d27af8aa-3a80-45e4-8730-7cd984bc69ff.png)

My idea is to use different quadrature rules in the meshes and thus, I need to identify each region by some IDs.

How best do I do that?

Thank you

**Comment by GiudGiud** [comment, marked as answer]
Hello

You could start with selective refinement using either of these mesh generators in 1D
https://mooseframework.inl.gov/source/meshgenerators/RefineBlockGenerator.html
https://mooseframework.inl.gov/source/meshgenerators/RefineSidesetGenerator.html

If you do want to do selective quadratures instead, then the syntax is shown here
https://mooseframework.inl.gov/source/actions/SetupQuadratureAction.html

For both options, you ll want to create different subdomains

Guillaume
